typing/typetexp.ml

```diff
@@ -506,35 +515,44 @@ end = struct
       { flavor; unbound_variable_policy; _ } env =
     let r = ref [] in
     TyVarMap.iter
-      (fun name (ty, loc) ->
-        if flavor = Unification || is_in_scope name then
-          let v = new_global_var (Jkind.Builtin.any ~why:Dummy_jkind) in
-          let snap = Btype.snapshot () in
-          if try unify env v ty; true with _ -> Btype.backtrack snap; false
-          then try
-            r := (loc, v, lookup_global name) :: !r
-          with Not_found ->
-            match unbound_variable_policy, Btype.is_Tvar ty with
-            | Open, _ | (Closed | Closed_for_upstream_compatibility), false ->
-              let jkind = Jkind.Builtin.any ~why:Dummy_jkind in
-              let v2 = new_global_var jkind in
-              r := (loc, v, v2) :: !r;
-              add name v2 jkind
-            | Closed, true ->
-              raise(Error(loc, env,
-                          Unbound_type_variable (Pprintast.tyvar_of_name name,
-                                                 get_in_scope_names (),
-                                                 None)))
-            | Closed_for_upstream_compatibility, true ->
-              raise(Error(loc, env,
-                          Unbound_type_variable (Pprintast.tyvar_of_name name,
+      (fun name (ty, loc, s) ->
```

Comment:
A large part of the diff below are due to indentation - I think they should be reverted to highlight the real changes.